Libraries/Components/Touchable/TouchableOpacity.js

```diff
@@ -105,12 +106,17 @@ var TouchableOpacity = React.createClass({
   },
 
   touchableHandleActivePressOut: function() {
-    this.setOpacityTo(1.0);
+    var childStyle = this.refs[CHILD_REF].props.style;
+
+    if (typeof childStyle === 'number') {
+      childStyle = StyleSheetRegistry.getStyleByID(childStyle);
+    }
+
+    this.setOpacityTo(childStyle && childStyle.opacity || 1);
```

Comment:
You cannot use || here. Otherwise if opacity is set to 0, it's going to be 1. You need childStyle.opacity === undefined ? 1 : childStyle.opacity